cmake/Gems.cmake

```diff
@@ -234,6 +234,14 @@ function(ly_enable_gems)
         set(ENABLED_GEMS ${store_temp}) # restore value of ENABLED_GEMS just in case...
     endif()
 
+    # Remove any version specifiers before looking for variants
+    # e.g. "Atom==1.2.3" becomes "Atom"
+    foreach(gem_name_with_version_specifier IN LISTS ly_enable_gems_GEMS)
+        o3de_get_name_and_version_specifier(${gem_name_with_version_specifier} gem_name spec_op spec_version)
+        list(APPEND GEM_NAMES "${gem_name}")
+    endforeach()
+    set(ly_enable_gems_GEMS ${GEM_NAMES})
```

Comment:
This should unset `GEM_NAMES` before the `foreach` loop.
When functions in cmake are invoked, it's values are visible to any nested calls to other functions within it.
https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#variables

What this means is that if the caller has a GEM_NAMES variable in scope, then a copy of it would be used to initialize this function call scope version of `GEM_NAMES` and that value would be appended to.